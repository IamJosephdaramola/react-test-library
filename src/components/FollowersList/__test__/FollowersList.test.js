import { render, screen, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import FollowersList from "../FollowersList";

const MockedFollowersList = () => (
  <BrowserRouter>
    <FollowersList />
  </BrowserRouter>
);

describe("FollowersList", () => {
  afterEach(cleanup);

  test("should take a snapshot of the FollowersList", () => {
    const { asFragment } = render(<MockedFollowersList />);

    expect(asFragment(<MockedFollowersList />)).toMatchSnapshot();
  });

  test("should render a follower", async () => {
    render(<MockedFollowersList />);

    const divElement = await screen.findByTestId("follower-item-0");
    expect(divElement).toBeInTheDocument();
  });

  //   test("should render multiple  followers", async () => {
  //     render(<MockedFollowersList />);

  //     const followersElement = await screen.findAllByTestId(/follower-item/i);

  //     expect(followersElement.length).toBe(5);
  //   });
});
