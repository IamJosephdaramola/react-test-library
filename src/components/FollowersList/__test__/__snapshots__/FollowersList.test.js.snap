// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowersList should take a snapshot of the FollowersList 1`] = `
<DocumentFragment>
  <div
    class="followerslist-container"
  >
    <div />
    <div
      class="todo-footer"
    >
      <a
        href="/"
      >
        Go Back
      </a>
    </div>
  </div>
</DocumentFragment>
`;
